### DISCUSSION TITLE
Recompiling frontend JS

### DISCUSSION THREAD

**Discussion by braindeaf** [loco-rs/loco#439] [General]
This is the first time I've had a play with Loco, I've been working with Rails since 2006 and trying to find my feet with Rust so this seem liked a great project to play with early on. The first thing I've tried to do is tweak the frontend React root component but it doesn't recompile automatically or on restart of the application. Am I missing something in the documentation?

Many thanks

RobL

**Comment by jondot** [comment]
Hello Rubyfriend :)

Just top of my head, rebuilding JS should be done by your JS bundler (e.g. Vite) as you change frontend code.
Then, to cycle your Loco app as code changes in it (Rust or otherwise), you can use something like `cargo-watch`, then:

```
$ cargo-watch -x check  -s 'cargo loco start'
```

You can reduce cycle time by removing `-x check` for example, and play with some `cargo-check` flags.

Two things to keep in mind since you're coming from Ruby and Rails:

1. Rust is a compiled langauge, and so "hot swapping" Rust code _should be very tricky_. I believe Bevy does that but in all honesty implementing such a feature is very, very tricky (as will be with any other compiled langauge). This is why instead of tackling this technological burden, we keep it simple and encourage using an external watcher.
2. Starting and restarting a Loco app is VERY VERY fast (this is mostly due to Rust, and partly due to minimal optimizations we made on the boot hot path). Faster than starting a Rails app, faster than starting a real-world Node app. You can restart it for every code change and it'll feel seamless and sharp.
